src/components/shared/Terminal.tsx

```diff
@@ -64,18 +66,25 @@ function Terminal(prop: {
   }
 
   function executeTouch(path: string, directory = false): string[] {
+    // Make a copy to avoid mutating the original (it's a state variable)
     const fileSystemCopy = _.cloneDeep(prop.fileSystem);
     const cwdCopy = fileSystemCopy.getFileSystemObjectFromPath(
       prop.currentWorkingDirectory.path
     );
     let newFileName = '';
     let dir;
 
+    // If the path is absolute, find the directory to create the file in
+    // Otherwise, use the current working directory
     if (path.includes('/')) {
```

Comment:
Refactor/backlog - Could we make this into a helper function to make it more readable / reusable